### PR TITLE
Fix double wrapping({:ok,{:ok, value}}) problem on tuple policy with source KV

### PR DESCRIPTION
**Pull request by altuntasfatih**
Heys,
We used `the tuple policy ` to properly return the data to the caller for fine-gradient error handling.
However, we faced this issue on the KV source. 
It wraps data by the ok tuple again without controlling whether the data is the ok tuple or not. So it wraps, then data becomes` {: ok,{: ok, value}}.`

I have added the following line code in two asterisks to the KV fetch function. It does not wrap again if the data is an ok tuple.
```
 case Map.fetch(batch, id) do
          :error -> {:error, "Unable to find id #{inspect(id)}"}
          {:ok, {:error, reason}} -> {:error, reason}
          ** {:ok, {:ok, item}} -> {:ok, item} **
          {:ok, item} -> {:ok, item}
        end
```

Actually, The code checks the error tuple but not the ok tuple case. I guess that was forgotten somehow.

 In addition,
 
  *  I have replaced the then/2 function with an anonymous function to fix the pipeline
  *  I have replaced the use Mix.Config statement with import Config to remove the deprecated warning.

### CODE REVIEW

**Comment by altuntasfatih** [comment]
Could you review it? @akoutmos, @benwilson512, @jadlr, @mbuhot

File: config/config.exs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This file is responsible for configuring your application
 # and its dependencies with the aid of the Mix.Config module.
-use Mix.Config
+import Config
```
Comment (kdawgwilk) [inline config/config.exs:3]: Seems unrelated to the change being proposed, you could probably remove these to make the changeset smaller

Comment (altuntasfatih) [reply config/config.exs]: I have reverted them to make the changeset smaller. @kdawgwilk could you review again?

File: test/dataloader/ecto_test.exs [hunk 2]
```diff
@@ -38,13 +38,13 @@ defmodule Dataloader.EctoTest do
...
     |> where(^Enum.to_list(args))
-    |> then(fn user ->
+    |> (fn user ->
       if is_nil(sort_by) or is_nil(sort_order) do
         user
       else
         order_by(user, {^sort_order, ^sort_by})
       end
-    end)
+    end).()
```
[the remark this answers is not in the document]
Comment (altuntasfatih) [reply test/dataloader/ecto_test.exs:47]: The current pipeline of this repo is failing because of the Kernel.then/2 function(error: undefined function then/2). 
The existing pipeline has jobs for Elixir 1.1X and OTP 22. 
Whereas The Kernel.then/2 function was added with Elixir.1.12
https://github.com/absinthe-graphql/dataloader/actions/runs/2652520264/jobs/4121762498.

I have replaced the then/2 function with an anonymous function to fix current the pipeline.

@benwilson512

File: test/dataloader_test.exs [hunk 3]
```diff
@@ -232,6 +242,26 @@ defmodule DataloaderTest do
...
+      result =
+        loader
+        |> Dataloader.load(:test, :results, :succes_data)
+        |> Dataloader.run()
+        |> Dataloader.get(:test, :results, :succes_data)
+
+      assert result == {:ok, :value}
+    end
+
+    test "get/4 returns an {:ok, value} tuple when data is error tuple", %{loader: loader} do
```
Comment (mbuhot) [inline test/dataloader_test.exs:255]:
```suggestion
    test "get/4 returns an {:error, value} tuple when data is error tuple", %{loader: loader} do
```

Comment (altuntasfatih) [reply test/dataloader_test.exs]: @mbuhot, could you approve?

**Comment by mbuhot** [comment]
It would be helpful to update the moduledoc: https://github.com/absinthe-graphql/dataloader/blob/master/lib/dataloader/kv.ex#L5-L7 to clarify the contract for the `load_function`.

Something along the lines of:

> You must supply a function that accepts a batch key and list of ids, and returns a map of values keyed by id. Values may optionally be returned as `:ok` / `:error` tuples to indicate success of the operation.

**Comment by altuntasfatih** [comment]
Could you approve again to run workflows ?

File: config/config.exs (hunk 1 above)
Comment (kdawgwilk) [reply config/config.exs]: My approval won't count unfortunately since I am not a maintainer

File: test/dataloader_test.exs (hunk 3 above)
Comment (benwilson512) [reply test/dataloader_test.exs:255]: @altuntasfatih: This test looks like it has the wrong name, I think @mbuhot is right.

Comment (altuntasfatih) [reply test/dataloader_test.exs:255]: yeah @benwilson512, I agree it is wrong. 
Actually, I had changed it from here(sending the suggested commit) then I overrode it from the local commit mistakenly.

Comment (altuntasfatih) [reply test/dataloader_test.exs]: I have just amended it.

**Comment by altuntasfatih** [comment]
@benwilson512, there is a conflict between .tool-versions and pipeline. The pipeline works on Elixir-1.10-11&OTP-22-23, whereas the .tool-versions works on Elixir-1.12-13&OTP-24. This causes problems. We should upgrade the pipeline.

For instance, the mix format works on both versions differently. So that is the reason the[ pipeline says the files are not formatted.](https://github.com/absinthe-graphql/dataloader/actions/runs/3204353858/jobs/5274310772#step:9:13)

At local, OTP24-Elixir1.13 says the format is okay, but Otp 23-elixir.1.11 says it is not formatted.

**Comment by altuntasfatih** [comment]
@benwilson512, there is a conflict between .tool-versions and pipeline. The pipeline works on OTP-22-23, whereas the .tool-versions works on OTP 24. This causes problems we should upgrade the pipeline if our version is OTP 24.
So that is the reason the pipeline says the files are not formatted

**Comment by altuntasfatih** [comment]
I have updated the pipeline as well.

**Comment by altuntasfatih** [comment]
Is there anyone who checks or approves this pr?

File: test/dataloader_test.exs [hunk 4]
```diff
@@ -8,7 +8,11 @@ defmodule DataloaderTest do
     users: [
       [id: "ben", username: "Ben Wilson"],
       [id: "bruce", username: "Bruce Williams"]
-    ]
+    ],
+    results: %{
+      succes_data: {:ok, :value},
```
Comment (benwilson512) [inline test/dataloader_test.exs:13]:
```suggestion
      success_data: {:ok, :value},
```

File: test/dataloader_test.exs [hunk 5]
```diff
@@ -232,6 +242,26 @@ defmodule DataloaderTest do
       assert result == {:ok, [id: "ben", username: "Ben Wilson"]}
     end
 
+    test "get/4 returns an {:ok, value} tuple when data is value tuple", %{loader: loader} do
+      result =
+        loader
+        |> Dataloader.load(:test, :results, :succes_data)
```
Comment (benwilson512) [inline test/dataloader_test.exs:248]:
```suggestion
        |> Dataloader.load(:test, :results, :success_data)
```

File: test/dataloader_test.exs [hunk 6]
```diff
@@ -232,6 +242,26 @@ defmodule DataloaderTest do
...
+        |> Dataloader.run()
+        |> Dataloader.get(:test, :results, :succes_data)
```
Comment (benwilson512) [inline test/dataloader_test.exs:250]:
```suggestion
        |> Dataloader.get(:test, :results, :success_data)
```

**Review by benwilson512 — CHANGES_REQUESTED** [review]

File: test/dataloader_test.exs (hunk 4 above)
Comment (altuntasfatih) [reply test/dataloader_test.exs]: i have changed it

File: test/dataloader_test.exs (hunk 5 above)
Comment (altuntasfatih) [reply test/dataloader_test.exs]: i have changed it

File: test/dataloader_test.exs (hunk 6 above)
Comment (altuntasfatih) [reply test/dataloader_test.exs]: i have changed it

**Comment by altuntasfatih** [comment]
I have amended it. Could you check again @benwilson512 ?

**Comment by benwilson512** [comment]
@altuntasfatih the primary blocker here is that this is technically a breaking change. I'm not clear whether this was a bug or just bad design, but if anyone has code out there that relies on this it will break. We are already kicking off a 2.0 version now so I think we can include that with this.
